src/kotlin/example/spring/src/main/kotlin/com/planetnine/sessionless/controllers/SessionlessController.kt

```diff
@@ -0,0 +1,161 @@
+                    val usersString = usersFile.readText()
+                    val json = JSONObject(usersString)
+                    users = json
+                    json.getString(userUUID)
+                }
+            } catch (_: Exception) {
+                null
+            }
+        }
+    }
+
+    @GetMapping("/")
+    fun index(req: HttpServletRequest): String {
+        return "Hello, World!"
+    }
+
+
+    private var currentPrivateKey = ""
+    private fun webSignature(req: HttpServletRequest, message: String): MessageSignature? {
+        val privateKeyString = req.session.getAttribute("user") as? String
+            ?: return null
+        currentPrivateKey = privateKeyString
+        val privateKey = privateKeyString.toECPrivateKey()
+        return sessionless.sign(message, privateKey)
+    }
+
+    private suspend fun handleWebRegistration(req: HttpServletRequest): RegisterResponse {
+        val userUUID = sessionless.generateUUID()
+        val pair = sessionless.generateKeysAsync(KeyAccessInfo(userUUID))
+        val simple = pair.toECHex()
+        req.session.setAttribute("user", simple.privateKey)
+        saveUser(userUUID, simple.publicKey)
+        return RegisterResponse(userUUID, "Welcome to this example!")
+    }
+
+    data class RegisterResponse(val uuid: String, val welcomeMessage: String)
+    data class RegisterReqBody(
+        val signature: MessageSignature?, val publicKey: String?,
+        val enteredText: String?, val timestamp: Long?,
+    )
+
+    @PutMapping("/register")
+    fun register(
+        req: HttpServletRequest,
+        @RequestBody body: RegisterReqBody
+    ): RegisterResponse? {
+        val signature = body.signature
+            ?: return runBlocking(Dispatchers.IO) { handleWebRegistration(req) }
+        val publicKey = body.publicKey ?: return null
+        val timestamp = body.timestamp ?: return null
+        val enteredText = body.enteredText ?: return null
+
+        val message = JSONObject().apply {
+            put("publicKey", publicKey)
+            put("enteredText", enteredText)
+            put("timestamp", timestamp)
+        }.toString()
+
+        val verified = sessionless.verify(message, publicKey, signature)
+        if (!verified) {
+            println("Signature verification failed!")
+            return null
+        }
+
+        val uuid = sessionless.generateUUID()
+        saveUser(uuid, publicKey)
+        println("Registered user with UUID: $uuid")
+        return RegisterResponse(uuid, "Welcome to this example!")
+    }
+
+    data class CoolStuffResponse(val doubleCool: String?, val error: String?)
+    data class CoolStuffReqBody(
+        val userUUID: String, val signature: MessageSignature?,
+        val timestamp: Long, val coolness: String,
+    )
+
+    @PutMapping("/cool-stuff")
+    fun coolStuff(req: HttpServletRequest, @RequestBody body: CoolStuffReqBody): CoolStuffResponse {
+        val publicKey = getUserPublicKey(body.userUUID)
```

Comment:
So we standardized on `uuid` for the request bodies (and `pubKey`). I think it's probably worth it to just find and replace all the `userUUID`s with `uuid`, but if you want to keep `userUUID` on internal stuff that's ok, but anything that clients send will be just `uuid`.